include/pybind11/numpy.h

```diff
@@ -42,6 +44,30 @@ class array; // Forward declaration
 
 PYBIND11_NAMESPACE_BEGIN(detail)
 
+// Main author of this class: jbms@
+template <typename T>
+class LazyInitializeAtLeastOnceDestroyNever {
+public:
+    template <typename Initialize>
```

Comment:
Document that you expect the GIL to be held, or in any case that calls are serialized. (Otherwise the mutating accesses would cause races.)

Reply:
Done: 1ce27154d655dd768ddd855394f30969c476b9fb